src/TestLogger/Core/TestCaseNameParser.cs

```diff
@@ -54,175 +54,41 @@ public static ParsedName Parse(string fullyQualifiedName)
-                if (step == NameParseStep.FindNamespace)
+                var method = new Regex(@"^([a-z1-9_.]{1,})\.([a-z1-9_.]{1,})\.(.{1,})$", regexOptions); // This one picks up anything with just method parameters or no method params.
+                if (method.IsMatch(fullyQualifiedName))
                 {
-                    metadataNamespaceName = string.Join(string.Empty, output);
+                    var matches = method.Matches(fullyQualifiedName);
+                    metadataNamespaceName = matches[0].Groups[1].Value;
+                    metadataTypeName = matches[0].Groups[2].Value;
+                    metadataMethodName = matches[0].Groups[3].Value;
                 }
-                else if (step == NameParseStep.FindType)
+
+                var classData = new Regex(@"^([a-z1-9_.]{1,})\.([a-z1-9_.]{1,}\(.{0,}\))\.(.{1,})$", regexOptions); // This one picks up anything with class data and maybe method data
+                if (classData.IsMatch(fullyQualifiedName))
                 {
-                    metadataTypeName = string.Join(string.Empty, output);
+                    var matches = classData.Matches(fullyQualifiedName);
+                    metadataNamespaceName = matches[0].Groups[1].Value;
+                    metadataTypeName = matches[0].Groups[2].Value;
+                    metadataMethodName = matches[0].Groups[3].Value;
                 }
             }
-            catch (Exception)
-            {
-                // On exception, wipe out the type name
-                metadataTypeName = string.Empty;
-                metadataNamespaceName = string.Empty;
-            }
-            finally
+
+            if (string.IsNullOrWhiteSpace(metadataNamespaceName)
+                || string.IsNullOrWhiteSpace(metadataTypeName)
+                || string.IsNullOrWhiteSpace(metadataMethodName))
             {
-                // If for any reason we don't have a Type Name or Namespace then we fall back on our
-                // safe option and notify the user
-                if (string.IsNullOrWhiteSpace(metadataNamespaceName) && string.IsNullOrWhiteSpace(metadataTypeName))
-                {
-                    metadataNamespaceName = TestCaseParserUnknownNamespace;
-                    metadataTypeName = TestCaseParserUnknownType;
-                    metadataMethodName = fullyQualifiedName;
-                    Console.WriteLine(TestCaseParserErrorTemplate, fullyQualifiedName, metadataNamespaceName, metadataTypeName, metadataMethodName);
-                }
-                else if (string.IsNullOrWhiteSpace(metadataNamespaceName))
-                {
-                    metadataNamespaceName = TestCaseParserUnknownNamespace;
-                    Console.WriteLine(TestCaseParserErrorTemplate, fullyQualifiedName, metadataNamespaceName, metadataTypeName, metadataMethodName);
-                }
+                metadataNamespaceName = TestCaseParserUnknownNamespace;
```

Comment:
Will "unknown" namespace be reported in the report xml? Will that make the reporting unreadable or garble the statistics in reporting? Is there a way for us to report a default fallback as if no parsing has happened? Or, any other better way?

`Console.WriteLine` will be invasive and pollute the CI logs. Should we consider using Trace Logger from the test platform (IIRC it was called `EqtTrace` or similar)? This will write the logging line to vstest diagnostics logs if enabled.

Reply:
Yeah, these still get  to the loggers. So you might get output like:

 `<test-case name="ExampleTest2(False,4.5m,4.6m,(4.5, False))" fullname="UnknownNamespace.UnknownType.MyNamespace.ExampleTest.ExampleTest2(False,4.5m,4.6m,(4.5, False))" methodname="MyNamespace.ExampleTest.ExampleTest2(False,4.5m,4.6m,(4.5, False))" classname="UnknownType" ... />`

Yeah, we have gotten several comments about the console output being annoying, particularly when this was happening to many tests. I'm not sure its important to have that output, or not more than once per run. If we can't parse it they see in the report. And we can make sure docs for this are discoverable.